### DISCUSSION TITLE
Multisite analysis

### DISCUSSION THREAD

**Discussion by Fahad021** [NREL/REopt-API-Analysis#10] [8 - API: Application Programming Interface] [answered]
Hi,
I am new in using ReOpt API. For multi-site analysis, the scnearios.csv is parsed and created as a "list_of_posts" (post requests). This list is processed by a for loop which iterates over the lists and the rate limit for POST is 200 per hour. The GET results has no rate limit associated with it. So for 1000 sites, we would need atleast 5 hours.
Am i correct in my assumption?

**Comment by Fahad021** [comment]
@NLaws @jmpohl

**Comment by NLaws** [comment, marked as answer]
@Fahad021 you are correct. We hope to increase the limit as we can. 

More details than you probably want: each POST to the job endpoint can use up to 3 CPUs and 4 GB RAM at its peak. Many of our users wish to run 1,000's of jobs per day, and so to allow equal access we have set the hourly limit to 200 jobs per hour.  A typical job takes 45 seconds to a minute. So if five users submit 200 jobs per hour we have 1,000 jobs per hour, that is about 17 jobs per minute, consuming up to 51 cores and 68 GB of RAM. 

It is important to note that these numbers are ballpark averages. Some jobs are faster, and the most complex ones take 5 minutes or more.  All jobs sit in a queue waiting on CPUs/workers to become available. So in some cases submitting 200 jobs in one hour does not mean that all those jobs will complete within an hour.

**Comment by NLaws** [comment]
FYI one can check their rate-limit usage using the [information on developer.nrel.gov](https://developer.nrel.gov/docs/rate-limits/)
